Automattic/pocket-casts-ios
# [Deselecting chapters] Add header to toggle chapter selection

| 📘 Part of: #1396 | Depends on #1431 |
|:---:|:---:|

Toggle the chapter selection/deselection.

At the moment, the chapter selection/deselection is done when the chapter is selected/deselected. We're still debating about this so I won't change this behavior for now. Also, this is not the final design.

https://github.com/Automattic/pocket-casts-ios/assets/7040243/bce97720-723e-41b4-9274-a6e5014eb813

These are podcasts with chapters you can use to test:

* Clublife: https://pca.st/rLzZCv
* ATP: https://pca.st/accidentaltech

## To test

2. Run the app
3. Go to Profile > Settings > Beta Features > `deselectChapters` and turn it on
4. Play any episode with chapters
5. Go to the Chapters tab
6. ✅ You should see a header indicating the number of chapters and a button to "Skip chapters"
7. Tap "Skip chapters"
8. ✅ You can now select/deselect chapters
9. Tap "Done"
10. ✅ The number of chapters and hidden chapters should be correctly updated and hidden chapters don't appear on the listing
11. Leave some chapters deselected
12. Go to Profile > Settings > Beta Features > `deselectChapters` and turn it off
13. Open Chapters tab again
14. ✅ All chapters should be there and there shouldn't be a header

## Checklist
